lib/rucio/richclient.py

```diff
@@ -0,0 +1,283 @@
+            if re.match(str(cfg_forced_modules), module_name):
+                return True
+        return False
+
+    # creating log
+    if not logger_name:
+        if not module_name:
+            logger_name = 'usr'
+        else:
+            logger_name = module_name.split('.')[-1]
+    logger = logging.getLogger(logger_name)
+
+    # extracting the log level
+    if not logger_level:
+        logger_level = logging.INFO
+        if verbose:
+            logger_level = logging.DEBUG
+
+        # overriding by the config
+        cfg_levels = (logging.DEBUG, logging.INFO, logging.WARNING, logging.ERROR)
+        for level in cfg_levels:
+            cfg_opt = 'forceloglevel' + logging.getLevelName(level)
+            if _force_cfg_log_level(cfg_opt):
+                logger_level = level
+
+    # setting the log level
+    logger.setLevel(logger_level)
+
+    # preferred logger handling
+    def add_handler(logger: logging.Logger,
+                    console: Optional[Console] = None,
+                    verbose: bool = False) -> None:
+
+        def time_formatter(timestamp: datetime) -> Text:
+            formatted_time = timestamp.strftime("%Y-%m-%d %H:%M:%S")
+            milliseconds = timestamp.microsecond // 1000
+            return Text(f"[{formatted_time},{milliseconds:03d}]")
+
+        console = console or Console()
+        highlighter = ReprHighlighter()
+        handler = RichHandler(rich_tracebacks=True, markup=True, show_path=verbose, show_time=verbose, console=console, tracebacks_width=min(console.width, MAX_TRACEBACK_WIDTH),
+                              tracebacks_word_wrap=True, log_time_format=time_formatter, highlighter=highlighter)
+        logger.addHandler(handler)
+
+    # setting handler and formatter
+    if not logger.handlers:
+        console = console or Console()
+        add_handler(logger, console, verbose)
+
+    return logger
+
+
+def generate_table(table_format: box.Box, headers: list[str], rows: Sequence[Union[tuple[Any, Any], list[Any]]],
+                   col_alignments: Optional[list[JustifyMethod]] = None, show_header: bool = True,
+                   row_styles: Optional[list[str]] = None) -> Table:
+    """
+    Generates a rich Table object from given input rows.
+    :param table_format: style of the table
+    :param headers: list of headers
+    :param rows: list of rows
+    :param col_alignments: list of column alignments
+    :param show_header: boolean to show header
+    :param row_styles: list of row styles
+    :returns: rich Table object
+    """
+    table = Table(box=table_format, show_header=show_header)
+    rich_widgets = (Text, Tree, Table)
+    if row_styles is None:
+        row_styles = ['none', 'dim']
+    table.row_styles = row_styles
+
+    if not col_alignments and len(rows) > 0:
+        # auto-detect on first row, numerical values on the right
+        col_alignments = ['right' if str(value).isnumeric() else 'left' for value in rows[0]]
```

Comment:
You’re using `value` here and `elem` below. Probably better to stick to one, and likely `column` or `col` so that it ties well with the term ‘row’.